src/vulkan-renderer/input/keyboard_mouse_data.cpp

```diff
@@ -1,41 +1,60 @@
 #include "inexor/vulkan-renderer/input/keyboard_mouse_data.hpp"
 
+#include <GLFW/glfw3.h>
+
+#include <shared_mutex>
+
 namespace inexor::vulkan_renderer::input {
 
+KeyboardMouseInputData::KeyboardMouseInputData() {
+    std::unique_lock lock(m_input_mutex);
+    /// We store the pressed keys in a vector of size GLFW_KEY_LAST.
+    m_pressed_keys.resize(GLFW_KEY_LAST);
+    m_pressed_mouse_buttons.resize(GLFW_KEY_LAST);
+}
+
```

Comment:
You can do `std::vector<bool> m_pressed_keys{GLFW_KEY_LAST}` on the members to avoid the constructor.

Reply:
Done.

Reply:
This can't be done like this (see follow up posts)